pkg/events/queue/queue_mem_list.go

```diff
@@ -0,0 +1,114 @@
+package queue
+
+import (
+	"container/list"
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"sync"
+)
+
+type EventQueueMem struct {
+	cache                *list.List
+	mutex                sync.Mutex
+	maxAmountOfEvents    int // number of cached events
+	EventsCacheMemSizeMB int
+}
+
+func (q *EventQueueMem) String() string {
+	return fmt.Sprintf("In-Memory Event Queue (Size = %d MB)", q.EventsCacheMemSizeMB)
+}
+
+func (q *EventQueueMem) Setup() error {
+	q.maxAmountOfEvents = q.getQueueSizeInEvents()
+	q.cache = new(list.List)
+
+	return nil
+}
+
+// Enqueue pushes an event into the queue (may block until queue is available)
+func (q *EventQueueMem) Enqueue(evt *external.Event) {
+	if evt == nil {
+		return
+	}
+
+	// "spin" until cache length is smaller then maxAmountOfEvents (blocking)
+	for {
+		q.mutex.Lock() // can't be RO as PushBack will update the list
+		if q.cache.Len() > q.maxAmountOfEvents {
+			q.mutex.Unlock() // release mutex to allow concurrency
+			continue
+		}
+		break
+	}
+
+	q.cache.PushBack(*evt)
+	q.mutex.Unlock()
+	evt = nil
+}
+
+// Dequeue pops an event from the queue
+func (q *EventQueueMem) Dequeue(blocking bool) *external.Event {
+
+	// "spin" until cache length is bigger than 0
+	for {
+		q.mutex.Lock() // can't be RO as Remove() will update the list
+		if q.cache.Len() == 0 {
+			q.mutex.Unlock() // release mutex to allow concurrency
+			if !blocking {
+				return nil // return nil if non blocking
+			}
+			continue // continue until there is an event in the queue if blocking
+		}
+		break
+	}
+
+	e := q.cache.Front()
+	event := e.Value.(external.Event)
```

Comment:
I know we will be safe as we are enqueueing always objects of type `external.Event`, nevertheless I am always a fan of assertion checking: 
```
event, ok := e.Value.(external.Event)
if !ok {
...
}
```